src/api/data/tree/importPlan.ts

```diff
@@ -189,54 +185,35 @@ const replaceRefWithId =
       Object.entries(record).map(([k, v]) => [k, v === `@${ref.refId}` ? ref.id : v])
     ) as SObjectTreeInput;
 
-export const validatePlanContents =
-  (logger: Logger) =>
-  async (planPath: string, planContents: unknown): Promise<DataPlanPartFilesOnly[]> => {
-    const childLogger = logger.child('validatePlanContents');
-    const planSchema = path.join(
-      path.dirname(fileURLToPath(import.meta.url)),
-      '..',
-      '..',
-      '..',
-      '..',
-      'schema',
-      'dataImportPlanSchema.json'
-    );
+// eslint-disable-next-line no-underscore-dangle
+export function _validatePlanContents(logger: Logger, planPath: string, planContents: unknown): DataImportPlanArray {
+  const childLogger: Logger = logger.child('validatePlanContents');
 
-    const val = new SchemaValidator(childLogger, planSchema);
-    try {
-      await val.validate(planContents as AnyJson);
-      const output = planContents as DataPlanPartFilesOnly[];
-      if (hasRefs(output)) {
-        childLogger.warn(
-          "The plan contains the 'saveRefs' and/or 'resolveRefs' properties.  These properties will be ignored and can be removed."
-        );
-      }
-      if (!hasOnlySimpleFiles(output)) {
-        throw messages.createError('error.NonStringFiles');
-      }
-      return planContents as DataPlanPartFilesOnly[];
-    } catch (err) {
-      if (err instanceof Error && err.name === 'ValidationSchemaFieldError') {
-        throw messages.createError('error.InvalidDataImport', [planPath, err.message]);
-      } else if (err instanceof Error) {
-        throw SfError.wrap(err);
-      }
-      throw err;
+  const parseResults = DataImportPlanArraySchema.safeParse(planContents);
+
+  if (parseResults.error) {
+    throw messages.createError('error.InvalidDataImport', [
+      planPath,
+      parseResults.error.issues.map((e) => e.message).join('\n'),
+    ]);
+  }
+  const parsedPlans: DataImportPlanArray = parseResults.data;
+
+  for (const parsedPlan of parsedPlans) {
+    if (parsedPlan.saveRefs !== undefined || parsedPlan.resolveRefs !== undefined) {
+      childLogger.warn(
+        "The plan contains the 'saveRefs' and/or 'resolveRefs' properties. These properties will be ignored and can be removed."
```

Comment:
Shane mentioned that I should actually print something to the console here instead of logging it where nobody will see it. The next step for me is to investigate the cleanest way to do that.